.markdownlint/customRule.js

```diff
@@ -15,11 +15,10 @@ module.exports = {
           "lineNumber": heading.lineNumber,
           "detail": "First heading should be '# Changelog'.",
         });
-      } else if (heading.tag === "h2" && !/^## \[(\d\.\d{1,9}\.\d{1,3}(-[.A-Za-z0-9]+)?)|Unreleased\]| - \d{4}-\d{2}-\d{2}$/.test(heading.line)) {
-        // every second heading should be a version number, then a dash and a space, and then a date
+      } else if (heading.tag === "h2" && !/^## (\[\d\.\d{1,9}\.\d{1,3}(-[.A-Za-z0-9]+)?\] - \d{4}-\d{2}-\d{2})|(\[Unreleased\])$/.test(heading.line)) {
```

Comment:
unrelated but this fixes the linter to support `[Unreleased]`, which is valid per keep a changelog spec